singer-commons/src/main/java/com/pinterest/singer/loggingaudit/client/utils/KafkaUtils.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2019 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pinterest.singer.loggingaudit.client.utils;
+
+
+import com.pinterest.singer.thrift.configuration.KafkaProducerConfig;
+
+import com.google.common.base.Joiner;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import kafka.cluster.Broker;
+import kafka.utils.ZkUtils;
+import org.I0Itec.zkclient.ZkClient;
+import org.I0Itec.zkclient.ZkConnection;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.producer.KafkaProducer;
+
+import java.util.Properties;
+import java.util.UUID;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.config.internals.BrokerSecurityConfigs;
+import org.apache.kafka.common.network.ListenerName;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import scala.Tuple2;
+import scala.collection.Seq;
+
+public class KafkaUtils {
+
+  private static Map<String, ZkUtils> zkUtilsMap = new HashMap();
+
+  public static KafkaProducer<byte[], byte[]> createKafkaProducer(KafkaProducerConfig config, String name) {
+    String brokerList = Joiner.on(',').join(config.getBrokerLists());
+    Properties properties = new Properties();
+    properties.put(
+        ProducerConfig.CLIENT_ID_CONFIG, name + "_" + UUID.randomUUID());
+    properties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList);
+    properties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, config.getKeySerializerClass());
+    properties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, config.getValueSerializerClass());
+
+    if (config.isTransactionEnabled()) {
```

Comment:
a lot of code is duplicated with the code in singer/src/main/java/com/pinterest/singer/utils/KafkaUtils.java. can we avoid the duplication?

Reply:
The Producer creation code is borrowed from the singer/src/main/java/com/pinterest/singer/utils/KafkaUtils.java.  To void de-duplicate, it requires moving packages/files in Singer module to Singer-commons module.  Given the current logging-audit branch is quite away from the master branch,  I feel it is safer to do the refactoring and avoiding duplicate AFTER  the logigng-audit branch is merged into the master branch.

Reply:
We should do the refactoring upfront, instead of after logging-audit branch is merged into the master branch.

Reply:
refactored KafkaUtils.java in the update of the PR to maintain only one piece of code.